cumulusci/robotframework/Salesforce.py

```diff
@@ -499,6 +525,115 @@ def salesforce_insert(self, obj_name, **kwargs):
         return res["id"]
 
+    def _salesforce_generate_object(self, obj_name, **fields):
+        obj = {"attributes": {"type": obj_name}}  # Object type to create
+        obj.update(fields)
+        return obj
+
+    random_string_generator = StringGenerator(String().generate_random_string)
+    faker_template_library = FakerTemplateLibrary()
+
+    def salesforce_collection_generate(self, obj_name, number_to_create, **fields):
+        """Returns an array of dictionaries with template-formatted arguments appropriate for a Collection Insert.
+            Use ``{{number}}`` to represent the unique index of the row in the list of rows, ``{{random_str}}`` to represent a random string
+            and a string that consists of just ``{{number}}`` to generate an actual integer (as opposed to a string-encoded number)
+
+            For example:
+
+                | @{objects} =  Salesforce Collection Generate  Contact  3
+                | ...  FirstName=User {{number}}
+                | ...  LastName={{random_str}}
+                | ...  Age={{number}}
+
+            Which would generate:
+
+                | [{'FirstName': 'User 0', 'LastName': 'u4PNlGUD', 'type': 'Contact', 'Age': '0'},
+                |  {'FirstName': 'User 1', 'LastName': 'Kyd4PC5x', 'type': 'Contact', 'Age': '1'},
+                |  {'FirstName': 'User 2', 'LastName': 'n4EGs4Vp', 'type': 'Contact', 'Age': '2'}]
+
+            Jinja2 Expression Syntax is allowed so computed templates like this are also allowed: {{1000 + number}}
+
+            Templates can also be based on faker patterns like those described here:
+
+            https://faker.readthedocs.io/en/master/providers.html
+
+            Most examples can be pasted into templates verbatim:
+
+        @{objects}=  Salesforce Collection Generate  Contact  200
+        ...  FirstName={{fake.first_name}}
+        ...  LastName={{fake.last_name}}
+        ...  MailingStreet={{fake.street_address}}
+        ...  MailingCity=New York
+        ...  MailingState=NY
+        ...  MailingPostalCode=12345
+        ...  Email={{fake.email(domain="salesforce.com")}}
+
+           """
+        objs = []
+
+        def format_str(value, i):
+            if isinstance(value, str):
+                value = Template(value).render(
+                    number=i,
+                    random_str=self.random_string_generator,
+                    fake=self.faker_template_library,
+                )
+
+            return value
+
+        for i in range(int(number_to_create)):
+            formatted_fields = {
+                name: format_str(value, i) for name, value in fields.items()
+            }
+            newobj = self._salesforce_generate_object(obj_name, **formatted_fields)
+            objs.append(newobj)
+
+        return objs
+
+    def salesforce_collection_insert(self, objects):
+        """Inserts up to 200 records that were created with Salesforce Collection Generate.
```

Comment:
instead of hard-coding 200 in several places, would it make sense to make this a constant defined at the start of the file? What is the chance that the API limit will change in the foreseeable future?

Reply:
I did this but it isn't really practical to do it in docstrings. (It's possible but the code gets pretty ugly).

Reply:
I think I'll leave it as it is. The constraint is validated client side (I think you suggested that) so the limit is fixed until we update the code which we'll do if Salesforce changes their limits.